src/storage/check.rs

```diff
@@ -1,118 +1,49 @@
 
-use std::marker::PhantomData;
-use std::ops::{Deref, DerefMut, Not};
+use std::ops::{Deref, Not};
 
 use hibitset::BitSet;
 
 use storage::{AntiStorage, MaskedStorage};
-use world::EntityIndex;
-use {Component, DistinctStorage, Index, Join, Storage, UnprotectedStorage};
+use {Component, DistinctStorage, Index, Join, Storage};
 
-/// A storage type that iterates entities that have
-/// a particular component type, but does not return the
-/// component.
-pub struct CheckStorage<'a, T, D> {
+/// Allows iterating over a storage without borrowing the storage itself.
+pub struct CheckStorage {
```

Comment:
> If you need to insert/remove components from the storage, this allows you to do that with a single loop rather than loop over all of them, find the ones you need to remove, store them in a Vec, then loop over those and remove them.

What is the performance difference with this single loop over the loop/deleteloop style?